sdk/include/aditof/adsd_errors.h

```diff
@@ -0,0 +1,361 @@
+/*
+ * BSD 3-Clause License
+ *
+ * Copyright (c) 2023, Analog Devices, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef ADSD3500_DEFS
+#define ADSD3500_DEFS
+
+#include <cstdint>
+#include <string>
+#include <unordered_map>
+
+namespace aditof {
+
+/**
+ * @class ADSDErrors
+ * @brief Error codes from the ADSD3500, ADSD3100, ADSD3030
+ * Usage:
+ * Included with aditof/camera.h
+ * For example: 
+ *    ADSDErrors err;
+ *    LOG(INFO) << ADSDErrors::ADSD3500_STATUS_INVALID_MODE;
+ *    LOG(INFO) << err.GetString(ADSDErrors::ADSD3500, 0x0001);
+ */
+class ADSDErrors {
+  public:
+    /**
+     * @brief The hardware types for the error for GetString.
+     */
+    enum ADSD3xxx { ADSD3500, ADSD3100, ADSD3030 };
```

Comment:
I would use 'enum class' instead of just 'enum'. It makes the type more strong (otherwise compiler will replace these names with values such as 0, 1, 2).